### ISSUE TITLE
Gemini AC doesn't complete in certain cases

### ISSUE THREAD

**Issue by LaurenWags** [brave/brave-browser#17460]
## Description 

Auto contribute using Gemini wallet doesn't complete when you close the browser before getting a transaction status back from Gemini.

## Steps to Reproduce

1. Clean profile, non-Japan region, staging environment, reconcile-interval=(some number of minutes). Enable Gemini in brave://flags and relaunch as necessary.
2. Enable Rewards
3. Select Verify Wallet and login with Gemini KYC'd account
4. Visit a mix of verified sites, (note, you may wish to adjust your AC budget to be higher than the default amount so that all expected sites get a contribution). Try to visit at least one of each of the following
- Uphold verified creator
- Gemini verified creator
- bitFlyer verified creator
- Brave verified, but not KYC'd, creator
- non-verified creator

5. Wait for auto contribute to begin
6. Once I saw `Will fetch transaction status after 3 minutes`, I closed the browser
7. Waited at least 5 min (since retry was supposed to occur in 3 min per the above logs), re-launched
8. Watch logs

## Actual result:

AC does not complete (BAT was deducted from Gemini acct)

```
[90216:259:0812/084515.590229:VERBOSE1:contribution.cc(739)] Retrying contribution (6b7481e3-2a00-4cb7-b3ca-004c0a167e3e) on step STEP_START
[90216:259:0812/084515.605728:VERBOSE5:ledger_impl.cc(119)] 
[ REQUEST ]
> URL: https://payment.rewards.bravesoftware.com/v1/orders/9........7/transactions/anonymousCard
> Method: POST
> Content: {"kind":"anonymous-card","paymentId":"4.......b","transaction":"eyJo.........
[ RESPONSE - OnRequest ]
> Url: https://payment.rewards.bravesoftware.com/v1/orders/9..........7/transactions/anonymousCard
> Result: Failure
> HTTP Code: 500
> Body: {"message":"Error creating anon card transaction: error submitting anon card transaction","code":500}
[90216:259:0812/084515.965086:VERBOSE9:logging_util.cc(137)] 
[ RESPONSE HEADERS ]
> Url: https://payment.rewards.bravesoftware.com/v1/orders/9............7/transactions/anonymousCard
[90216:259:0812/084515.966186:INFO:post_transaction_anon.cc(108)] Internal server error
[90216:259:0812/084515.967248:INFO:contribution_anon_card.cc(53)] Problem sending transaction
[90216:259:0812/084515.968321:INFO:sku_transaction.cc(118)] Transaction for order failed 9.........7
[90216:259:0812/084515.969305:INFO:sku_common.cc(61)] Order status was not updated
[90216:259:0812/084515.970280:INFO:contribution_sku.cc(184)] SKU was not processed
```

Example | Example
----- | -----
<img width="404" alt="Screen Shot 2021-08-12 at 8 47 17 AM" src="https://user-images.githubusercontent.com/28145373/129227486-047a67cb-ac6e-4afa-99cd-ca8b55e83165.png"> | <img width="931" alt="Screen Shot 2021-08-12 at 8 47 39 AM" src="https://user-images.githubusercontent.com/28145373/129227492-e64e19ec-e04e-44d3-9f3f-b752616e0331.png">

## Expected result:
AC should continue to check for Gemini txn status and then complete

## Reproduces how often: 

easily

## Brave version (brave://version info)

1.29.57 (Beta)

## Version/Channel Information:

- Can you reproduce this issue with the current release? n/a
- Can you reproduce this issue with the beta channel? Yes
- Can you reproduce this issue with the nightly channel? Yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:

cc @jumde @Miyayes @brave/legacy_qa

**Comment by LaurenWags** [comment]
@jumde mentioned this might be resolved with https://github.com/brave/brave-browser/issues/17290

**Comment by LaurenWags** [comment]
Also tested this with a tip as well.

Steps:
1. Clean profile, non-Japan region, staging environment. Enable Gemini in brave://flags and relaunch as necessary.
2. Enable Rewards
3. Select Verify Wallet and login with Gemini KYC'd account
4. Tip a verified, KYC'd Gemini creator (ex. bat community.org):

<img width="1189" alt="Screen Shot 2021-08-24 at 3 52 26 PM" src="https://user-images.githubusercontent.com/28145373/130682093-b7edd601-ec38-4777-9b2c-8d125870096a.png">

5. Wait for tip to be created in logs:

```
[7066:259:0824/155234.013470:VERBOSE1:contribution.cc(362)] Creating contribution for wallet type gemini (amount: 1, type: ONE_TIME_TIP)
[7066:259:0824/155234.016678:VERBOSE1:contribution.cc(91)] Queue timer set for 0 s
[7066:259:0824/155234.017344:VERBOSE1:database_server_publisher_banner.cc(142)] Server publisher banner not found
[7066:259:0824/155234.018548:INFO:post_transaction_gemini.cc(103)] Initiating gemini transaction to: 60e86f02-81f8-40fd-86f6-c2d6be298b37for 0.95
[7066:259:0824/155234.018605:VERBOSE5:ledger_impl.cc(119)] 
[ REQUEST ]
> URL: https://api.qa6.aurora7.net/v1/payments/pay
> Method: POST
> Content Type: application/json; charset=utf-8
[7066:259:0824/155234.203201:VERBOSE6:logging_util.cc(136)] 
[ RESPONSE - OnRequest ]
> Url: https://api.qa6.aurora7.net/v1/payments/pay
> Result: Success
> HTTP Code: 200
> Body: {"result":"OK","tx_ref":"9b32cc4b-5210-4252-af0d-3afa8d9ffca3","amount":0.95,"currency":"BAT","destination":"60e86f02-81f8-40fd-86f6-c2d6be298b37","status":"Pending"}
[7066:259:0824/155234.203295:VERBOSE9:logging_util.cc(137)] 
[ RESPONSE HEADERS ]
> Url: https://api.qa6.aurora7.net/v1/payments/pay
[7066:259:0824/155234.204286:INFO:post_transaction_gemini.cc(116)] Gemini transaction successful
[7066:259:0824/155234.204354:VERBOSE1:post_transaction_gemini.cc(129)] Transfer Pending
[7066:259:0824/155234.204399:VERBOSE1:gemini_transfer.cc(75)] Will fetch transaction status after 3 minutes
```

6. Once I saw `Will fetch transaction status after 3 minutes`, I closed the browser
7. Waited at least 3 min and restarted the browser
8. I can see in the UI that my balance has decreased by the 95% portion of the tip, but the tip isn't mentioned:

Example | Example | Example
----- | ----- | -----
<img width="684" alt="Screen Shot 2021-08-24 at 4 01 03 PM" src="https://user-images.githubusercontent.com/28145373/130682111-8009193c-b5da-4484-ac89-4a5a8beb489a.png"> | <img width="402" alt="Screen Shot 2021-08-24 at 4 01 09 PM" src="https://user-images.githubusercontent.com/28145373/130682115-65aef706-e4c0-4c7a-b77f-4b24caf0e5b3.png"> | <img width="422" alt="Screen Shot 2021-08-24 at 4 01 23 PM" src="https://user-images.githubusercontent.com/28145373/130682119-19f6f29c-ba90-4767-b567-164d7fca98ad.png">

<details>
<summary>Logs</summary>

```
[ RESPONSE - OnRequest ]
> Url: https://api.qa6.aurora7.net/v1/payments/pay
> Result: Success
> HTTP Code: 200
> Body: {"result":"OK","tx_ref":"6a2e3dcb-e34e-4002-95a6-df067dbf1fca","amount":0.95,"currency":"BAT","destination":"60e86f02-81f8-40fd-86f6-c2d6be298b37","status":"Pending"}
[7172:259:0824/155630.341044:VERBOSE9:logging_util.cc(137)] 
[ RESPONSE HEADERS ]
> Url: https://api.qa6.aurora7.net/v1/payments/pay
[7172:259:0824/155630.341678:INFO:post_transaction_gemini.cc(116)] Gemini transaction successful
[7172:259:0824/155630.341726:VERBOSE1:post_transaction_gemini.cc(129)] Transfer Pending
[7172:259:0824/155630.341760:VERBOSE1:gemini_transfer.cc(75)] Will fetch transaction status after 3 minutes
```

```
[7172:259:0824/155930.342935:VERBOSE1:gemini_transfer.cc(86)] Fetching transaction status 1 time
[7172:259:0824/155930.343989:VERBOSE5:ledger_impl.cc(119)] 
[ REQUEST ]
> URL: https://api.qa6.aurora7.net/v1/payment/60df8caa-795b-4eef-a8bf-585e3799d454/6a2e3dcb-e34e-4002-95a6-df067dbf1fca
> Method: GET
> Content Type: application/json; charset=utf-8
[7172:259:0824/155930.344778:VERBOSE6:logging_util.cc(136)] 
[ RESPONSE - OnRequest ]
> Url: https://api.qa6.aurora7.net/v1/balances
> Result: Success
> HTTP Code: 200
> Body: [{"type":"exchange","currency":"BAT","amount":"10008.15","available":"10008.15","availableForWithdrawal":"10008.15"}]
[7172:259:0824/155930.344852:VERBOSE9:logging_util.cc(137)] 
[ RESPONSE HEADERS ]
> Url: https://api.qa6.aurora7.net/v1/balances
[7172:259:0824/155930.504967:VERBOSE6:logging_util.cc(136)] 
[ RESPONSE - OnRequest ]
> Url: https://api.qa6.aurora7.net/v1/payment/60df8caa-795b-4eef-a8bf-585e3799d454/6a2e3dcb-e34e-4002-95a6-df067dbf1fca
> Result: Success
> HTTP Code: 200
> Body: {"tx_ref":"6a2e3dcb-e34e-4002-95a6-df067dbf1fca","amount":0.95,"currency":"BAT","destination":"60e86f02-81f8-40fd-86f6-c2d6be298b37","status":"Pending"}
[7172:259:0824/155930.505075:VERBOSE9:logging_util.cc(137)] 
[ RESPONSE HEADERS ]
> Url: https://api.qa6.aurora7.net/v1/payment/60df8caa-795b-4eef-a8bf-585e3799d454/6a2e3dcb-e34e-4002-95a6-df067dbf1fca
[7172:259:0824/155930.505134:VERBOSE1:get_transaction_gemini.cc(88)] Transfer Status: Pending
[7172:259:0824/155930.506246:INFO:gemini_transfer.cc(104)] Number of active retry timers: 0
[7172:259:0824/155930.506329:VERBOSE1:gemini_transfer.cc(75)] Will fetch transaction status after 6 minutes
```

```
[7172:259:0824/160530.509091:VERBOSE1:gemini_transfer.cc(86)] Fetching transaction status 2 time
[7172:259:0824/160530.510284:VERBOSE5:ledger_impl.cc(119)] 
[ REQUEST ]
> URL: https://api.qa6.aurora7.net/v1/payment/60df8caa-795b-4eef-a8bf-585e3799d454/6a2e3dcb-e34e-4002-95a6-df067dbf1fca
> Method: GET
> Content Type: application/json; charset=utf-8
[7172:259:0824/160531.139521:VERBOSE6:logging_util.cc(136)] 
[ RESPONSE - OnRequest ]
> Url: https://api.qa6.aurora7.net/v1/payment/60df8caa-795b-4eef-a8bf-585e3799d454/6a2e3dcb-e34e-4002-95a6-df067dbf1fca
> Result: Success
> HTTP Code: 200
> Body: {"tx_ref":"6a2e3dcb-e34e-4002-95a6-df067dbf1fca","amount":0.95,"currency":"BAT","destination":"60e86f02-81f8-40fd-86f6-c2d6be298b37","status":"Completed","timestampms":1629835390361}
[7172:259:0824/160531.139644:VERBOSE9:logging_util.cc(137)] 
[ RESPONSE HEADERS ]
> Url: https://api.qa6.aurora7.net/v1/payment/60df8caa-795b-4eef-a8bf-585e3799d454/6a2e3dcb-e34e-4002-95a6-df067dbf1fca
[7172:259:0824/160531.139692:VERBOSE1:get_transaction_gemini.cc(88)] Transfer Status: Completed
[7172:259:0824/160531.140600:INFO:gemini_transfer.cc(104)] Number of active retry timers: 0
[7172:259:0824/160531.140664:VERBOSE1:gemini.cc(215)] Gemini transfer fee timer set for 122 s
```

Note - at this point the tip was deducted a second time:
<img width="1121" alt="Screen Shot 2021-08-24 at 4 08 18 PM" src="https://user-images.githubusercontent.com/28145373/130682878-0f916e97-4944-48e1-82e2-69931b827f29.png">

```
[7172:259:0824/160733.148491:INFO:post_transaction_gemini.cc(103)] Initiating gemini transaction to: 60edbc1b-eaa7-4aa0-afb8-a87a89634c55for 0.05
[7172:259:0824/160733.148607:VERBOSE5:ledger_impl.cc(119)] 
[ REQUEST ]
> URL: https://api.qa6.aurora7.net/v1/payments/pay
> Method: POST
> Content Type: application/json; charset=utf-8
[7172:259:0824/160733.330349:VERBOSE6:logging_util.cc(136)] 
[ RESPONSE - OnRequest ]
> Url: https://api.qa6.aurora7.net/v1/payments/pay
> Result: Success
> HTTP Code: 200
> Body: {"result":"OK","tx_ref":"52d16cc2-d11f-4e25-8f8a-71f95b53a6de","amount":0.05,"currency":"BAT","destination":"60edbc1b-eaa7-4aa0-afb8-a87a89634c55","status":"Pending"}
[7172:259:0824/160733.330454:VERBOSE9:logging_util.cc(137)] 
[ RESPONSE HEADERS ]
> Url: https://api.qa6.aurora7.net/v1/payments/pay
[7172:259:0824/160733.331560:INFO:post_transaction_gemini.cc(116)] Gemini transaction successful
[7172:259:0824/160733.331640:VERBOSE1:post_transaction_gemini.cc(129)] Transfer Pending
[7172:259:0824/160733.331696:VERBOSE1:gemini_transfer.cc(75)] Will fetch transaction status after 3 minutes
```

```[7172:259:0824/161033.334332:VERBOSE1:gemini_transfer.cc(86)] Fetching transaction status 1 time
[7172:259:0824/161033.335535:VERBOSE5:ledger_impl.cc(119)] 
[ REQUEST ]
> URL: https://api.qa6.aurora7.net/v1/payment/60df8caa-795b-4eef-a8bf-585e3799d454/52d16cc2-d11f-4e25-8f8a-71f95b53a6de
> Method: GET
> Content Type: application/json; charset=utf-8
[7172:259:0824/161033.774213:VERBOSE6:logging_util.cc(136)] 
[ RESPONSE - OnRequest ]
> Url: https://api.qa6.aurora7.net/v1/payment/60df8caa-795b-4eef-a8bf-585e3799d454/52d16cc2-d11f-4e25-8f8a-71f95b53a6de
> Result: Success
> HTTP Code: 200
> Body: {"tx_ref":"52d16cc2-d11f-4e25-8f8a-71f95b53a6de","amount":0.05,"currency":"BAT","destination":"60edbc1b-eaa7-4aa0-afb8-a87a89634c55","status":"Pending"}
[7172:259:0824/161033.774278:VERBOSE9:logging_util.cc(137)] 
[ RESPONSE HEADERS ]
> Url: https://api.qa6.aurora7.net/v1/payment/60df8caa-795b-4eef-a8bf-585e3799d454/52d16cc2-d11f-4e25-8f8a-71f95b53a6de
[7172:259:0824/161033.774312:VERBOSE1:get_transaction_gemini.cc(88)] Transfer Status: Pending
[7172:259:0824/161033.774973:INFO:gemini_transfer.cc(104)] Number of active retry timers: 0
[7172:259:0824/161033.775019:VERBOSE1:gemini_transfer.cc(75)] Will fetch transaction status after 6 minutes
```

```
[7172:259:0824/161633.775469:VERBOSE1:gemini_transfer.cc(86)] Fetching transaction status 2 time
[7172:259:0824/161633.776019:VERBOSE5:ledger_impl.cc(119)] 
[ REQUEST ]
> URL: https://api.qa6.aurora7.net/v1/payment/60df8caa-795b-4eef-a8bf-585e3799d454/52d16cc2-d11f-4e25-8f8a-71f95b53a6de
> Method: GET
> Content Type: application/json; charset=utf-8
[7172:259:0824/161634.212230:VERBOSE6:logging_util.cc(136)] 
[ RESPONSE - OnRequest ]
> Url: https://api.qa6.aurora7.net/v1/payment/60df8caa-795b-4eef-a8bf-585e3799d454/52d16cc2-d11f-4e25-8f8a-71f95b53a6de
> Result: Success
> HTTP Code: 200
> Body: {"tx_ref":"52d16cc2-d11f-4e25-8f8a-71f95b53a6de","amount":0.05,"currency":"BAT","destination":"60edbc1b-eaa7-4aa0-afb8-a87a89634c55","status":"Completed","timestampms":1629835990416}
[7172:259:0824/161634.212338:VERBOSE9:logging_util.cc(137)] 
[ RESPONSE HEADERS ]
> Url: https://api.qa6.aurora7.net/v1/payment/60df8caa-795b-4eef-a8bf-585e3799d454/52d16cc2-d11f-4e25-8f8a-71f95b53a6de
[7172:259:0824/161634.212394:VERBOSE1:get_transaction_gemini.cc(88)] Transfer Status: Completed
[7172:259:0824/161634.213498:INFO:gemini_transfer.cc(104)] Number of active retry timers: 0
```

</details>

If you wait long enough you will see that the tip (95% portion) gets deducted twice and the 5% fee gets deducted once:
<img width="1122" alt="Screen Shot 2021-08-24 at 4 14 57 PM" src="https://user-images.githubusercontent.com/28145373/130683720-d37e9b74-8cc9-49b6-8171-af8ec5f09711.png">

**Comment by Miyayes** [comment]
Resolved with server-side fix.
